lib/cli/src/commands/run/mod.rs

```diff
@@ -140,9 +143,81 @@ impl Run {
         #[cfg(not(feature = "sys"))]
         let engine = store.engine().clone();
 
-        let runtime =
-            self.wasi
-                .prepare_runtime(engine, &self.env, runtime, preferred_webc_version)?;
+        let registry_name = self
+            .env
+            .registry_public_url()?
+            .host_str()
+            .unwrap_or("unknown_registry")
+            .replace('.', "_");
+
+        // We don't have the bytes of the module yet, but we still want to have the
+        // package-capabilities cache be as close to an actual identifier as possible.
+        let package_cache_path = match &self.input {
+            PackageSource::File(f) => {
+                let full_path = f.canonicalize()?.to_path_buf();
+                let metadata = full_path
+                    .parent()
+                    .ok_or(anyhow!("No parent!"))?
+                    .metadata()?
+                    .modified()?;
+
+                let mut hash = twox_hash::XxHash64::default();
+                full_path.hash(&mut hash);
+                metadata.hash(&mut hash);
+
+                format!("path_{}.json", hash.finish())
+            }
+            PackageSource::Dir(f) => {
+                let full_path = f.canonicalize()?.to_path_buf();
+                let metadata = full_path.metadata()?.modified()?;
+
+                let mut hash = twox_hash::XxHash64::default();
+                full_path.hash(&mut hash);
+                metadata.hash(&mut hash);
+
+                format!("path_{}.json", hash.finish())
+            }
+            PackageSource::Package(p) => match p {
+                PackageSpecifier::Ident(id) => {
+                    format!("id_{id}.json")
```

Comment:
Should we only use the namespace + name here, and re-use across versions?

Would be sensible and more convenient IMO.

Reply:
[5b8321d](https://github.com/wasmerio/wasmer/pull/4975/commits/5b8321d098b02353e0a960f805851a1456a29338)